corporate/tests/test_stripe.py

```diff
@@ -4016,6 +4027,30 @@ class Row:
                     email_found = True
             self.assertEqual(row.email_expected_to_be_sent, email_found)
 
+    @mock_stripe()
+    def test_upgrade_pay_by_invoice(self, *mock: Mock) -> None:
+        hamlet = self.example_user("hamlet")
+        self.login_user(hamlet)
+        response = self.client_get("/upgrade/?setup_payment_by_invoice=true")
+        self.assert_in_success_response(["pay by card", "Send invoice"], response)
+
+        # Send invoice
+        response = self.client_billing_post(
+            "/billing/upgrade",
+            {
+                "billing_modality": "send_invoice",
+                "schedule": "annual",
+                "signed_seat_count": self.get_signed_seat_count_from_response(response),
+                "salt": self.get_salt_from_response(response),
+                "license_management": "manual",
+                "licenses": 40,
+            },
+        )
+        self.assert_json_success(response)
+
+        response = self.client_get("/upgrade/?setup_payment_by_invoice=true")
+        self.assert_in_success_response(["An invoice", "has been sent"], response)
```

Comment:
It would be really great to extend this commit with the end-to-end upgrade flow. Maybe worth doing in a follow-up commit, though, since this commit is really hard to work with on GitHub.